nix/inferno-ml/migrations/v1-create-tables.sql

```diff
@@ -92,5 +96,32 @@ create table if not exists evalinfo
   , cpu bigint not null
   );
 
+-- Because the `params` table references an array of model versions, and
+-- because Postgres does not natively support arrays of foreign keys, this
+-- trigger function checks that each array item in the `models` column is
+-- a valid `mversions` primary key and that the model has not been terminated
+create or replace function verifymvs()
+returns trigger as $$
+declare
+  mv int;
+begin
+  foreach mv in array new.models loop
+    if
+      not exists
+        ( select 1 from mversions
+          where id = mv
+            and terminated is null
+        )
+    then
+      raise exception 'ID % is not a valid model version primary key', mv;
+    end if;
+  end loop;
+  return new;
+end;
+$$ language plpgsql;
+
+create trigger "verify-param-models" before insert or update on params
```

Comment:
if going this route, a trigger is also needed to prevent modelversions which are referrenced by params from being deleted